### PR TITLE
Switch from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by cottsay**
This change moves all of the tests currently using Travis CI to use GitHub Actions. The jobs are largely one-to-one with the original Travis jobs, except that the rosdistro audit job was split into separate ROS 1 and ROS 2 jobs.

### CODE REVIEW

**Review by j-rivero — COMMENTED** [review]
Hard to check that the new CI matches somehow the travis one. All I can see is good, some changes and updates to Ubuntu / ROS distributions, Great work @cottsay . I left a couple of comments.

File: .github/actions/audit/action.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+name: ros_buildfarm audit
+description: Execute a ROS buildfarm audit jobs
+inputs:
+  config_url:
+    description: URL of the buildfarm configuration index
+    required: true
+    default: https://raw.githubusercontent.com/ros-infrastructure/ros_buildfarm_config/production/index.yaml
+  config_name:
```
[the remark this answers is not in the document]
Comment (cottsay) [reply .github/actions/audit/action.yaml]: Nice catch - dropped in dba08a0

File: .travis.yml [hunk 2]
```diff
@@ -1,544 +0,0 @@
-_ros_gpg_key_add_string: &rosgpgkeyadd |
```
[the remark this answers is not in the document]
Comment (cottsay) [reply .travis.yml:1]: I took a different approach here. The OSRF repositories, which were only being added to get catkin to list off the test results, have been dropped. You don't even need catkin to be installed to use those executables, so I just pull the latest noetic code for catkin and invoke the binary from there.

[the remark this answers is not in the document]
Comment (cottsay) [reply .travis.yml:1]: Here's where I do that: https://github.com/ros-infrastructure/ros_buildfarm/blob/dba08a0871bc0e7520f88e7c67dd459a2daddf77/.github/actions/setup/action.yaml#L11-L13

**Review by j-rivero — APPROVED** [review]
Good to go. pytest is complaining about a missing doc string in one of the test files for pulp, I think outside of the scope of this PR.

**Pull request merged**